src/main/kotlin/no/nb/bikube/newspaper/service/AxiellService.kt

```diff
@@ -98,6 +98,30 @@ class AxiellService  (
             .map { mapCollectionsObjectToGenericTitle(it) }
     }
 
+    fun getItemsByTitle(
+        titleCatalogId: String,
+        date: LocalDate,
+        isDigital: Boolean,
+        materialType: MaterialType
+    ): Flux<CatalogueRecord> {
+        return axiellRepository.getSingleCollectionsModel(titleCatalogId)
+            .flatMapIterable { it.getObjects() ?: emptyList() }
+            .flatMapIterable { getPartsFromTitle(it) }
+            .flatMapIterable { getYearWorksFromTitle(it) }
+            .filter { (_, manifestationPartsObject) -> filterByYear(manifestationPartsObject, date) }
+            .flatMapIterable { getItemsReferenceFromManifestation(it) }
+            .filter { (_, itemPartReference, _) -> filterByFormat(itemPartReference, isDigital) }
+            .map { (title, itemPartReference, date) ->
+                mapCollectionsPartsObjectToGenericItem(
+                    itemPartReference!!,
+                    titleCatalogId,
+                    title,
+                    materialType.value,
+                    date
+                )
+            }
```

Comment:
You can avoid using Pair / Triple with a little bit of Reactor nesting:

```suggestion
        return axiellRepository.getSingleCollectionsModel(titleCatalogId)
            .flatMapIterable { it.getObjects() ?: emptyList() }
            .flatMap { titleObject ->
                val title = titleObject.titleList?.first()?.title ?: ""
                Flux.fromIterable(titleObject.getParts() ?: emptyList())
                    .flatMapIterable { it.getPartRefs() }
                    .filter { manifestationPartsObject -> filterByYear(manifestationPartsObject, date) }
                    .flatMapIterable { it.getPartRefs() }
                    .filter { itemPartReference -> filterByFormat(itemPartReference.partsReference, isDigital) }
                    .map { itemPartReference ->
                        mapCollectionsPartsObjectToGenericItem(
                            itemPartReference.partsReference!!,
                            titleCatalogId,
                            title,
                            materialType.value,
                            date.toString()
                        )
                    }
            }
```

`title` is the only value you need to keep. The `date` value in the last Triple has to be equal to the method parameter.

If this suggestion is correct, you can also remove the 3 new `get...()` methods below in this file.

Reply:
Tested suggestion and it works like a charm 🤩